src/core_scan/Report_pro_findings.ml

```diff
@@ -0,0 +1,107 @@
+(*****************************************************************************)
+(* Purpose *)
+(*****************************************************************************)
+(** Support the reporting of pro findings
+
+  * Users want to know when a finding could only be found by the pro engine
+  * so that they can understand the value Semgrep provides. Additionally,
+  * we want to surface our "coolest" findings early to make sure users see
+  * them.
+
+  * To support this, we have a different type for the engine kind for matches,
+  * which includes a `PRO_SPECIFIC` option that reports when a finding could
+  * only be found using the pro engine. This option also allows us to highlight
+  * findings we think users should prioritize.
+  *)
+
+module FileSet = Set.Make (String)
+
+(*****************************************************************************)
+(* Helpers *)
+(*****************************************************************************)
+
+(* Check if the taint trace of the pattern crosses files *)
```

Comment:
In theory we should also report as pro-findings when traces cross functions.

Reply:
We do! I rearranged the functions to make this more visible